uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -67,18 +71,22 @@ define([
        * @return {Promise} Resolved with {Object} Updated Leave request
        */
       update: function (params) {
-        $log.debug('LeaveRequestAPI.update');
+        $log.debug('LeaveRequestAPI.update', params);
         var deferred = $q.defer();
 
         if (!params.id) {
           deferred.reject('id is mandatory field');
         }
 
-        deferred.resolve(this.sendPOST('LeaveRequest', 'create', params)
+        this.sendPOST('LeaveRequest', 'create', params)
           .then(function (data) {
+            if(data && data.is_error == 1){
```

Comment:
1. `if(data && !!data.is_error) {`
2. Leave a space before "{"
3. We should not send the error object via `reject`, instead we can just send  
`deferred.reject(data.error_message)`;